src/platform/i386/chal/chal_prot.h

```diff
@@ -0,0 +1,73 @@
+/******************************************************************************
+Filename    : chal_prot.h
+Author      : Runyu Pan
+Date        : ?/?/2018
+Licence     : GPL v2; see COPYING for details.
+Description : The Composite chal general prototype interface.
```

Comment:
Descriptions need to be worded carefully.

"The generic Composite Hardware Abstraction Layer (chal) interface that is separately implemented for each architecture."